src/neo4j_genai/filters.py

```diff
@@ -164,11 +164,11 @@ class ParameterStore:
     Determine the parameter name depending on a parameter counter
     """
 
-    def __init__(self):
-        self._counter = Counter()
-        self.params = {}
+    def __init__(self) -> None:
+        self._counter: Counter[str] = Counter()
+        self.params: dict[str, Union[str, int, list[Any]]] = {}
```

Comment:
I'd use `Any` as type for the dict value, it could potentially be a float or even a date (good question, we should test dates btw)

Reply:
If possible I'd avoid `Any` if we know it's a `float` or a date type

Reply:
c9081d159d46cdaddbf90285271128c4cf7dc91a